### PR TITLE
Allow delegating to non-public methods for AdditionalAnswers#delegatesTo

### PR DESCRIPTION
**Pull request by li-wjohnson**
Fixes an issue where doing "delegatesTo(new AnonymousClass() {})" would cause an IllegalAccessException, and other access-related issues.

(See https://github.com/mockito/mockito/pull/1535 - Sorry, wasn't aware that you can't change the PR source branch.)

### CODE REVIEW

File: src/test/java/org/mockito/internal/stubbing/defaultanswers/foopackage/ForwardsInvocationsTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+package org.mockito.internal.stubbing.defaultanswers.foopackage;
+
+import org.junit.Test;
+import org.mockito.internal.stubbing.defaultanswers.ForwardsInvocations;
+import org.mockitoutil.TestBase;
+
+import static org.junit.Assert.assertEquals;
+
+public class ForwardsInvocationsTest extends TestBase {
```
Comment (mockitoguy) [inline src/test/java/org/mockito/internal/stubbing/defaultanswers/foopackage/ForwardsInvocationsTest.java]: Do we need this test? Does this test improves coverage on top of the existing ForwardsInvocationsTest?

File: src/main/java/org/mockito/internal/stubbing/defaultanswers/ForwardsInvocations.java [hunk 2]
```diff
@@ -40,6 +40,11 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
             }
 
             Object[] rawArguments = ((Invocation) invocation).getRawArguments();
+            try {
+                delegateMethod.setAccessible(true);
```
Comment (mockitoguy) [inline src/main/java/org/mockito/internal/stubbing/defaultanswers/ForwardsInvocations.java]: Can you add an e2e user test? Can you put together a test class that shows that without this change Mockito indeed throws access exceptions?

**Comment by mockitoguy** [comment]
Please ignore my earlier comment, I can see the new test is doing what we need to do!

File: src/test/java/org/mockito/internal/stubbing/defaultanswers/foopackage/ForwardsInvocationsTest.java (hunk 1 above)
Comment (TimvdLippe) [reply src/test/java/org/mockito/internal/stubbing/defaultanswers/foopackage/ForwardsInvocationsTest.java]: The original problem was the inability to mock anonymous classes. In line 21 of this test, an anonymous class is created. I think this is therefore testing the appropriate codepath?

**Comment by mockitoguy** [comment]
I reworked the test a little bit and will merge once the build is happy. Thank you @li-wjohnson!

File: src/test/java/org/mockito/internal/stubbing/defaultanswers/foopackage/ForwardsInvocationsTest.java (hunk 1 above)
Comment (mockitoguy) [reply src/test/java/org/mockito/internal/stubbing/defaultanswers/foopackage/ForwardsInvocationsTest.java]: It is. Please disregard my comment.

Comment (TimvdLippe) [reply src/test/java/org/mockito/internal/stubbing/defaultanswers/foopackage/ForwardsInvocationsTest.java]: Yeah, our comments were cross-posted. I saw your reply the second I posted my comment 😛

**Pull request merged**